src/resource_manager/gateway.py

```diff
@@ -140,7 +146,7 @@ def _delete_resource(self, name: str) -> None:
             name=name,
         )
 
-    def gateway_address(self, name: str) -> Optional[str]:
+    def gateway_address(self, name: str) -> typing.Optional[str]:  # pragma: no cover
```

Comment:
Would you be able to help explain the reason for the `no cover` as a comment if possible please? ;D

Reply:
Yes, I put `no cover` here since this code can waits for up to 60 seconds and it would slow down unit tests quite a bit if we test all the branches of this method

Reply:
Can't you mock `time.time()` in your unittest with a side effect where you'll be able to control the effective waiting time ?

Reply:
I didn't know that this was something you can do, I've remove `no cover` here and I'll add tests for this method in https://github.com/canonical/gateway-api-integrator-operator/pull/18